WordPress/wordpress-playground
# Build: Source external deps from package.json

Follows up on
https://github.com/WordPress/wordpress-playground/pull/1589 and #1586

Treating package IDs matching `node_modules` as external works, but causes Rollup to substitute the module name with the absolute module path on the host machine, e.g. `import "ini"` becomes `import "/home/runner/work/wordpress-playground/wordpress-playground/node_modules/ini/lib/ini.js"` which is not what we want.

This PR sources the external packages names from the package.json file to both treat all these imports as external AND preserve their original import identifier. It also adds the required `package.json` building steps to the `@php-wasm/universal` package and makes the `@php-wasm/stream-compression` package public to ensure the published packages will reference the correct dependencies.

 ## Testing instructions

Run nx build php-wasm-node and confirm the resultingdist/packages/php-wasm/index.js file and other built npm-published modules do import external modules using their name, not their absolute path.
